src/p2G4_main.c

```diff
@@ -183,7 +183,8 @@ static void f_rx_search(uint d) {
     }
   }
 
-  fq_add(current_time + 1, Rx_Search, d);
+  bs_time_t expiry = BS_MIN(rx_status->scan_end + 1, rx_status->rx_s.abort.recheck_time);
```

Comment:
the time of the next event in the queue is not known until all events at the current time point have executed. There may be other devices that run after this one that will add a new event to the queue with an earlier timestamp than anything currently in the queue.

Reply:
Yep.. what I proposed here would require rx_search to be the last type of event in the queue, and probably then adding a recheck @+1 if there is other elements to trigger now in the queue.. which for mesh is a certainty.. so it would give nothing for mesh...